### ISSUE TITLE
train_mnist give different result between two runs

### ISSUE THREAD

**Issue by howard0su** [dmlc/mxnet#4306]
## Environment info
Operating System:
Ubuntu 14, Windows 10

Compiler:
gcc, visual studio 2015

Package used (Python/R/Scala/Julia):
Python

MXNet version:
MXNet commit hash (`git rev-parse HEAD`):
962271410059156180ab1d5e79b805e687512be9

If you are using python package, please provide
Python version and distribution:
2.7.6

## Minimum reproducible example
python train_mnist.py --network lenet --gpus 1
The accuracy doesn't make any sense. Interesting I repo this on Linux after i first noticed this on Windows like issue #1228 

## Steps to reproduce
I tried twice with the same command:
howardsu@DPHost:~/mxnet/example/image-classification$ python train_mnist.py --network lenet --gpus 1
INFO:root:start with arguments Namespace(batch_size=64, disp_batches=100, gpus='1', kv_store='device', load_epoch=None, lr=0.1, lr_factor=0.1, lr_step_epochs='10', model_prefix=None, mom=0.9, network='lenet', num_classes=10, num_epochs=20, num_examples=60000, num_layers=None, optimizer='sgd', test_io=0, top_k=0, wd=0.0001)
INFO:root:Start training with [gpu(1)]
INFO:root:Epoch[0] Batch [100]	Speed: 21407.75 samples/sec	Train-accuracy=0.116250
INFO:root:Epoch[0] Batch [200]	Speed: 22608.54 samples/sec	Train-accuracy=0.101250
INFO:root:Epoch[0] Batch [300]	Speed: 24133.26 samples/sec	Train-accuracy=0.101562
INFO:root:Epoch[0] Batch [400]	Speed: 27369.95 samples/sec	Train-accuracy=0.102813
INFO:root:Epoch[0] Batch [500]	Speed: 27600.49 samples/sec	Train-accuracy=0.102344
INFO:root:Epoch[0] Batch [600]	Speed: 27567.07 samples/sec	Train-accuracy=0.099687
INFO:root:Epoch[0] Batch [700]	Speed: 27607.53 samples/sec	Train-accuracy=0.102344
INFO:root:Epoch[0] Batch [800]	Speed: 27500.14 samples/sec	Train-accuracy=0.099844
INFO:root:Epoch[0] Batch [900]	Speed: 27448.59 samples/sec	Train-accuracy=0.103594

howardsu@DPHost:~/mxnet/example/image-classification$ python train_mnist.py --network lenet --gpus 1
INFO:root:start with arguments Namespace(batch_size=64, disp_batches=100, gpus='1', kv_store='device', load_epoch=None, lr=0.1, lr_factor=0.1, lr_step_epochs='10', model_prefix=None, mom=0.9, network='lenet', num_classes=10, num_epochs=20, num_examples=60000, num_layers=None, optimizer='sgd', test_io=0, top_k=0, wd=0.0001)
INFO:root:Start training with [gpu(1)]
INFO:root:Epoch[0] Batch [100]	Speed: 21860.08 samples/sec	Train-accuracy=0.465625
INFO:root:Epoch[0] Batch [200]	Speed: 23576.40 samples/sec	Train-accuracy=0.757188
INFO:root:Epoch[0] Batch [300]	Speed: 22909.50 samples/sec	Train-accuracy=0.813438
INFO:root:Epoch[0] Batch [400]	Speed: 23619.98 samples/sec	Train-accuracy=0.831406
INFO:root:Epoch[0] Batch [500]	Speed: 23110.21 samples/sec	Train-accuracy=0.852344
INFO:root:Epoch[0] Batch [600]	Speed: 22845.57 samples/sec	Train-accuracy=0.865781
INFO:root:Epoch[0] Batch [700]	Speed: 23093.27 samples/sec	Train-accuracy=0.871563
INFO:root:Epoch[0] Batch [800]	Speed: 23533.74 samples/sec	Train-accuracy=0.888437
INFO:root:Epoch[0] Batch [900]	Speed: 22846.45 samples/sec	Train-accuracy=0.898906

**Comment by mli** [comment]
try a smaller lr?

**Comment by howard0su** [comment]
I didn't change any settings default in the example. lr=0.1 by default. If we believe lr is too big, better we change the example code.
